internal/constants/scripts.go

```diff
@@ -66,140 +70,149 @@ function map_host_macs_to_interfaces() {
   for nic in $( ls $SYS_CLASS_NET_DIR )
   do
     mac=$(cat $SYS_CLASS_NET_DIR/$nic/address | tr '[:lower:]' '[:upper:]')
-    host_macs_to_hw_iface[$mac]=$nic
+    [[ -n "$mac" ]] && host_macs_to_hw_iface[$mac]=$nic
   done
 }
 
-function find_host_directory_by_mac_address() {
-  for d in $(ls -d ${NMCONNECTIONS_DIR}/host*)
-  do
+# Copy all nmstate files representing a host nic
+function copy_physical_nics_files() {
+  host_src_dir=$1
+  mapping_file=$2
+  host_work_dir=$3
+  for mac_address in $(cat $mapping_file | cut -d= -f1 | tr '[:lower:]' '[:upper:]');  do
+    host_nic="${host_macs_to_hw_iface[${mac_address}]:-}"
+    if [[  -n "$host_nic" ]]; then
+      logical_nic="${logical_mac_nic_map[${mac_address}]:-}"
+      if [[ -n "$(ls -A $host_src_dir/*.nmconnection )" ]] ; then
+        for nmfile in $host_src_dir/*.nmconnection ; do
+          filename=$(basename $nmfile)
+          prefix="${filename%%.*}"
+          extension="${filename#*.}"
+          [[ -n "$prefix" ]] && [[ "$prefix" == "$logical_nic" ]] &&  cp $nmfile $host_work_dir/${host_nic}.${extension} && rm -f $nmfile
+        done
+      fi
+    fi
+  done
+}
 
-    mapping_file="${d}/${MAC_NIC_MAPPING_FILE}"
+# If any nmstate files of the current host directory were copied, copy the remaining nmstate files
+function copy_other_files() {
+  host_src_dir=$1
+  host_work_dir=$2
+  if [ "$(ls -A $host_work_dir )" ] && [[ -n "$(ls -A ${host_src_dir}/*.nmconnection)" ]] ; then
+    for nmfile in ${host_src_dir}/*.nmconnection ; do
+      filename=$(basename $nmfile)
+      target_file=$host_work_dir/$filename
+
+      # If the same filename was not used already, then use it. Otherwise generate another name.
```

Comment:
Is the file name important? Does it still need to match the device name at this point?

Reply:
Not for files that don't match device names (example: bond)